### PR TITLE
Add tests for validations being done before committing the pending fed

### PR DESCRIPTION
**Pull request by julia-zack**

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java [hunk 1]
```diff
@@ -330,6 +343,25 @@ void voteCommitFederation_commit1MemberFederation_returnsInsufficientMembersResp
         assertEquals(FederationChangeResponseCode.INSUFFICIENT_MEMBERS.getCode(), firstVoteResult);
     }
 
+    @Test
+    void voteCommitFederation_withMismatchedHashes_returnsFederationNonExistentResponseCode() {
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java:347]: The return in this case is `PENDING_FEDERATION_MISMATCHED_HASH`.

File: rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java [hunk 2]
```diff
@@ -305,6 +305,19 @@ void voteAddFederatorPublicKey_whenAdding100Federators_returnsSuccessfulResponse
...
 
     // vote commit federation tests
+    @Test
+    void voteCommitFederation_withNullPendingFederation_returnsFederationNonExistentResponseCode() {
+        // Arrange
+        Keccak256 randomHash = Keccak256.ZERO_HASH;
+        ABICallSpec commitFederationAbiCallSpec = new ABICallSpec(FederationChangeFunction.COMMIT.getKey(), new byte[][]{ randomHash.getBytes() });
+
+        // Act
+        int result = federationSupport.voteFederationChange(firstAuthorizedTx, commitFederationAbiCallSpec, signatureCache, bridgeEventLogger);
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java:315]:
```suggestion
        int actualFedChangeResponseCode = federationSupport.voteFederationChange(firstAuthorizedTx, commitFederationAbiCallSpec, signatureCache, bridgeEventLogger);
```

Wdyt? I think `result` is too generic.

File: rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java [hunk 3]
```diff
@@ -330,6 +343,25 @@ void voteCommitFederation_commit1MemberFederation_returnsInsufficientMembersResp
...
     }
 
+    @Test
+    void voteCommitFederation_withMismatchedHashes_returnsFederationNonExistentResponseCode() {
+        // Arrange
+        voteAndAssertCreateEmptyPendingFederation();
+        int minMembersRequired = 2;
+        voteAndAssertAddFederatorPublicKeysToPendingFederation(minMembersRequired);
+
+        Keccak256 randomHash = Keccak256.ZERO_HASH;
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java:353]:
```suggestion
        Keccak256 wrongHash = Keccak256.ZERO_HASH;
```

I think `wrongHash` or `nonPendingFederationHash` suits better. Wdyt?

File: rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java (hunk 1 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java:347]: you right! Thanks

Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java]: updated

File: rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java (hunk 2 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java:315]: is called result in all the other tests, no big deal i think

File: rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java (hunk 3 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java:353]: Still named as `randomHash`

**Review by nathanieliov — APPROVED** [review]

**Review by apancorb — APPROVED** [review]

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**